nyc_taxis/test_procedures/default.json

```diff
@@ -293,5 +293,150 @@
+        },
+        {
+          "operation": {
+            "name": "delete-local-index",
+            "operation-type": "delete-index"
+          }
+        },
+        {
+          "operation": "restore-snapshot"
+        },
+        {
+          "operation": "default",
+          "warmup-iterations": 50,
+          "iterations": 100
+          {%- if not target_throughput %}
+          ,"target-throughput": 3
+          {%- elif target_throughput is string and target_throughput.lower() == 'none' %}
+          {%- else %}
+          ,"target-throughput": {{ target_throughput | tojson }}
+          {%- endif %}
+          {%-if search_clients is defined and search_clients %}
+          ,"clients": {{ search_clients | tojson}}
+          {%- endif %}
+        },
+        {
+          "operation": "range",
+          "warmup-iterations": 50,
+          "iterations": 100
+          {%- if not target_throughput %}
+          ,"target-throughput": 0.7
+          {%- elif target_throughput is string and target_throughput.lower() == 'none' %}
+          {%- else %}
+          ,"target-throughput": {{ target_throughput | tojson }}
+          {%- endif %}
+          {%-if search_clients is defined and search_clients %}
+          ,"clients": {{ search_clients | tojson}}
+          {%- endif %}
+        },
+        {
+          "operation": "distance_amount_agg",
+          "warmup-iterations": 50,
+          "iterations": 100
+          {%- if not target_throughput %}
+          ,"target-throughput": 2
+          {%- elif target_throughput is string and target_throughput.lower() == 'none' %}
+          {%- else %}
+          ,"target-throughput": {{ target_throughput | tojson }}
+          {%- endif %}
+          {%-if search_clients is defined and search_clients %}
+          ,"clients": {{ search_clients | tojson}}
+          {%- endif %}
+        },
+        {
+          "operation": "autohisto_agg",
+          "warmup-iterations": 50,
+          "iterations": 100
+          {%- if not target_throughput %}
+          ,"target-throughput": 1.5
+          {%- elif target_throughput is string and target_throughput.lower() == 'none' %}
+          {%- else %}
+          ,"target-throughput": {{ target_throughput | tojson }}
+          {%- endif %}
+          {%-if search_clients is defined and search_clients %}
+          ,"clients": {{ search_clients | tojson}}
+          {%- endif %}
+        },
+        {
+          "operation": "date_histogram_agg",
+          "warmup-iterations": 50,
+          "iterations": 100
+          {%- if not target_throughput %}
+          ,"target-throughput": 1.5
+          {%- elif target_throughput is string and target_throughput.lower() == 'none' %}
+          {%- else %}
+          ,"target-throughput": {{ target_throughput | tojson }}
+          {%- endif %}
+          {%-if search_clients is defined and search_clients %}
+          ,"clients": {{ search_clients | tojson}}
+          {%- endif %}
+        }
+      ]
```

Comment:
Can you consider putting the common queries into a separate file and having both test-procedures `append-no-conflicts` and `searchable-snapshot` share them to avoid duplication?  Also, please rename `searchable_snapshot` to `searchable-snapshot` for consistency.

Reply:
I noticed the obvious code duplication of the `target_throughput` section. 👍 I've changed them in a previous commit https://github.com/opensearch-project/opensearch-benchmark-workloads/pull/58/commits/2a7138d050b645a0204334ad9f0a252d2b494889, by using [macros](https://jinja.palletsprojects.com/en/3.1.x/templates/#macros) syntax, but I changed those back eventually and decided to create a separate PR to make the change throughout the whole code repository, because there are lots of occurrences of the duplication in other files. The duplication of `target_throughput` are introduced by this PR https://github.com/opensearch-project/opensearch-benchmark-workloads/pull/47, it's kind of a bad change.
Do you think it's reasonable that I keep the duplication as it is in this PR, and create another PR to resolve the duplication of `target_throughput` throughout the whole repository.

Reply:
Yes, I'd leave the duplication as is for now, to keep this PR focused on Searchable Snapshots.  Thank you for creating the other issue, which we can address separately.

Reply:
@gkamat Thank you for your time in the review! I can handle the code refactor in next week.